wire-ios-data-model/Source/ManagedObjectContext/Migration/2.112-2.113/OneOnOneConversationMigrationPolicy.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+// Up until model version 2.112, a user was related to their one on one
+// conversation via the `connection` relationship, ie `user.connection.conversation`
+// and inversely `conversation.connection.to`.
+
+class OneOnOneConversationMigrationPolicy: NSEntityMigrationPolicy {
+
+    override func createRelationships(
+        forDestination dInstance: NSManagedObject,
+        in mapping: NSEntityMapping,
+        manager: NSMigrationManager
+    ) throws {
+        try super.createRelationships(
+            forDestination: dInstance,
+            in: mapping,
+            manager: manager
+        )
+
+        guard dInstance.entity.name == ZMUser.entityName() else {
+            return
+        }
+
+        guard let sourceUser = manager.sourceInstances(
+            forEntityMappingName: mapping.name,
+            destinationInstances: [dInstance]
+        ).first else {
+            return
+        }
+
+        // There is an existing connection for this user.
+        if let connection = sourceUser.value(forKey: "connection") as? NSManagedObject {
+            guard
+                let sourceConversation = connection.value(forKey: "conversation") as? NSManagedObject,
+                let destinationConversation = manager.destinationInstances(
+                    forEntityMappingName: "ConversationToConversation",
+                    sourceInstances: [sourceConversation]
+                ).first
+            else {
+                return
+            }
+
+            dInstance.setValue(destinationConversation, forKey: "oneOnOneConversation")
```

Comment:
Can the `sourceConversation` be deleted after successfully moving to `oneOnOneConversation`?

Reply:
Do you mean the relationship between the source conversation and source connection? If so, I don't think we need to singe that relationship no longer exists in the destination model.

Reply:
Not sure myself 😬 let's keep it this way as it works